internal/kafka/internal/handlers/validation.go

```diff
@@ -370,3 +373,93 @@ func ValidateKafkaStorageSize(kafkaRequest *dbapi.KafkaRequest, kafkaUpdateReq *
 		return nil
 	}
 }
+
+func validateNoKafkaPromotionInProgress(kafkaRequest *dbapi.KafkaRequest) handlers.Validate {
+	return func() *errors.ServiceError {
+		// TODO is this check desirable? what happens if a promotion has previously failed
```

Comment:
See this comment

Reply:
I believe this check should be removed: we just need to check the promotion status and allow the promotion if it is different than 'promoting' and 'promoted'

Reply:
I agree.
For example, if a previous promotion has previously failed desired could be different than actual in the DB and no promotion being in progress at all, as it had failed.

I will remove this check and keep the check that checks the promoting state field

Reply:
> I believe this check should be removed: we just need to check the promotion status and allow the promotion if it is different than 'promoting' and 'promoted'

Just a minor comment about this. There is no `promoted` value at all for promoting state field

Reply:
Updated